### PR TITLE
update stationary constraints

### PR DESCRIPTION
**Pull request by akshay-krishnan**
Set the stationary constraints to Pose3() with a very small covariance. 
When creating PosePrior from constraint, these are set as HARD_CONSTRAINTs.

### CODE REVIEW

File: gtsfm/loader/hilti_loader.py [hunk 1]
```diff
@@ -112,6 +114,20 @@ def __load_constraints(self) -> Dict[Tuple[int, int], Constraint]:
...
+    def _update_stationary_constraints(self, constraints: Dict[Tuple[int, int], Constraint]) -> Dict[Tuple[int, int], Constraint]:
+        updated_constraints = {}
+        TRANSLATION_THRESHOLD = 0.02
+        ROTATION_THRESHOLD = np.deg2rad(0.5)
+        for (a, b), constraint in constraints.items():
+            if np.linalg.norm(Rot3.Logmap(constraint.aTb.rotation())) < ROTATION_THRESHOLD or np.linalg.norm(constraint.aTb.translation()) < TRANSLATION_THRESHOLD:
+                updated_constraints[(a, b)] = constraint
+                continue
+            
+            updated_constraint = Constraint(constraint.a, constraint.b, Pose3(), np.eye(6) * 1e-8, constraint.counts)
```
Comment (ayushbaid) [inline gtsfm/loader/hilti_loader.py:126]: Can you make it not smaller than the rig constraint? Also maybe move it to a global variable above.

File: gtsfm/loader/hilti_loader.py [hunk 2]
```diff
@@ -112,6 +114,20 @@ def __load_constraints(self) -> Dict[Tuple[int, int], Constraint]:
         # cast them to dictionary
         return {(constraint.a, constraint.b): constraint for constraint in constraints}
 
+    def _update_stationary_constraints(self, constraints: Dict[Tuple[int, int], Constraint]) -> Dict[Tuple[int, int], Constraint]:
+        updated_constraints = {}
+        TRANSLATION_THRESHOLD = 0.02
+        ROTATION_THRESHOLD = np.deg2rad(0.5)
+        for (a, b), constraint in constraints.items():
```
Comment (ayushbaid) [inline gtsfm/loader/hilti_loader.py:121]: Also please count and log how many cases we catch.

**Review by dellaert — APPROVED** [review]

File: gtsfm/loader/hilti_loader.py [hunk 3]
```diff
@@ -136,8 +152,8 @@ def is_higher_step_magnitude_greater(a, b, all_magnitudes, step_size):
             if abs(a - b) != 1:
                 filtered_constraints[(a, b)] = constraint
                 continue
-            # Do not include if a higher-step magnitude is greater (for steps of size 2 and 3)
-            if is_higher_step_magnitude_greater(a, b, constraint_magnitudes, 1) or is_higher_step_magnitude_greater(a, b, constraint_magnitudes, 2):
+            # Do not include if a higher-step magnitude is lesser (for steps of size 2 and 3)
+            if is_higher_step_magnitude_lesser(a, b, constraint_magnitudes, 1) or is_higher_step_magnitude_lesser(a, b, constraint_magnitudes, 2):
```
Comment (ayushbaid) [inline gtsfm/loader/hilti_loader.py:156]: oh was this the bug?

File: gtsfm/loader/hilti_loader.py [hunk 4]
```diff
@@ -45,9 +45,8 @@
 LIDAR_CONSTRAINTS_RELATIVE_PATH = "lidar/constraints.txt"
 IMAGES_FOLDER = "images"
 
-HARD_RELATIVE_POSE_PRIOR_SIGMA = np.eye(6) * 1e-3  # CAM_IMU_POSE_PRIOR_SIGMA in BA should have similar value
-SOFT_RELATIVE_POSE_PRIOR_SIGMA = np.eye(6) * 3e-2
-SOFT_ABSOLUTE_POSE_PRIOR_SIGMA = np.eye(6) * 3e-2
+HARD_POSE_PRIOR_COVARIANCE = np.eye(6) * 1e-8
```
Comment (ayushbaid) [inline gtsfm/loader/hilti_loader.py:48]: oh wow I was using sigma as covariance!

**Comment by ayushbaid** [comment]
On exp01 dataset, 4130/36915 constraints are classified stationary. Does this sound right to you?

**Pull request merged**